command/agent/config/config.go

```diff
@@ -181,6 +181,9 @@ type ExecConfig struct {
 func NewConfig() *Config {
 	return &Config{
 		SharedConfig: new(configutil.SharedConfig),
+		TemplateConfig: &TemplateConfig{
+			MaxConnectionsPerHost: DefaultTemplateConfigMaxConnsPerHost,
+		},
```

Comment:
Moving this here because now that we do check correctly below, the resulting `TemplateConfig` reverted back to being nil again when not set. So the default needs to be set in `NewConfig()` so it is in the base config we use when merging